src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java

```diff
@@ -221,41 +211,32 @@ public final void configure(Configuration config)
      */
     private void tryCopyProperty(String key, Object value, boolean recheck)
             throws CheckstyleException {
-        final BeanUtilsBean beanUtils = createBeanUtilsBean();
-
         try {
-            if (recheck) {
-                // BeanUtilsBean.copyProperties silently ignores missing setters
-                // for key, so we have to go through great lengths here to
-                // figure out if the bean property really exists.
-                final PropertyDescriptor descriptor =
-                        PropertyUtils.getPropertyDescriptor(this, key);
-                if (descriptor == null) {
-                    final String message = getLocalizedMessage(
-                        AbstractAutomaticBean.class,
-                        "AbstractAutomaticBean.doesNotExist", key);
-                    throw new CheckstyleException(message);
-                }
+            // BeanUtilsBean.copyProperties silently ignores missing setters
+            // for key, so we have to go through great lengths here to
+            // figure out if the bean property really exists.
+            if (recheck && PropertyUtils.getPropertyDescriptor(this, key) == null) {
+                throw new CheckstyleException(getLocalizedMessage(
+                    AbstractAutomaticBean.class,
+                    "AbstractAutomaticBean.doesNotExist", key));
             }
             // finally we can set the bean property
-            beanUtils.copyProperty(this, key, value);
+            createBeanUtilsBean().copyProperty(this, key, value);
         }
         catch (final InvocationTargetException | IllegalAccessException
                 | NoSuchMethodException exc) {
             // There is no way to catch IllegalAccessException | NoSuchMethodException
             // as we do PropertyUtils.getPropertyDescriptor before beanUtils.copyProperty,
             // so we have to join these exceptions with InvocationTargetException
             // to satisfy UTs coverage
-            final String message = getLocalizedMessage(
```

Comment:
this actually directReturn rule and kind of bug that this code is what we have...